oci/spec_opts_test.go

```diff
@@ -704,3 +707,75 @@ func TestWithoutMounts(t *testing.T) {
 		t.Fatalf("expected %+v, got %+v", expected, s.Mounts)
 	}
 }
+
+func TestWithWindowsDevice(t *testing.T) {
```

Comment:
nit.. windows only test .. without goos check.. should also work on other platforms.. but won't be used on other platforms

Reply:
It's testing code in _spec_opts.go_, and the test doesn't have any reliance on the _runtime_ OS, so I think this is the right place for the test. Why _wouldn't_ I want to cover cross-platform code in cross-platform tests?

As to whether `oci.WithWindowsDevice` itself should be in _spec_opts_windows.go_... My initial thought was "No, `oci.X` only needs to support the local runtime platform, it'll only be used for building local specs". Then @kevpar [made a good point about this](https://github.com/containerd/containerd/pull/6618#pullrequestreview-905555749) and [I flipped](https://github.com/containerd/containerd/pull/6618#issuecomment-1063928000) to "`oci` shouldn't be local-runtime-specific needlessly".

I figured having the code here is also consistent with, e.g., `oci.WithWindowsHyperV` being in _spec_opts.go_.

Reply:
all true was just a nit.. in recognizing the code paths for using WithWindowsDevice() at runtime are limited to _windows in files except for the test .. agree the location spec_opts.go made sense.. I expected for the test to follow suit with the use cases.. raised an eyebrow enough to suggest at least a // comment there and thought I'd ask the question.. are we considering adding `"windows": {??}` to other platform configs..